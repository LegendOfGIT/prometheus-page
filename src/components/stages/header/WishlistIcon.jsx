import React from 'react';
import PropTypes from 'prop-types';
import {MODULE_ID_WISHLIST} from '../../../constants';

const WishlistIcon = ({setActiveModule, wishlistHasItems}) => {
    return (
        <div className={'wishlist'}>
            <img className={'wishlist__icon'}
                 src={`/images/wishlist-${wishlistHasItems ? 'selected' : 'neutral'}.svg`}
                 onClick={() => {
                     setActiveModule(MODULE_ID_WISHLIST);
                 }}
            />
            <h5 className={'wishlist__title'}>Wishlist</h5>
        </div>)
        ;
};

WishlistIcon.defaultProps = {
    wishlistHasItems: false
};

WishlistIcon.propTypes = {
    setActiveModule: PropTypes.func.isRequired,
    wishlistHasItems: PropTypes.bool.isRequired
};

export default WishlistIcon;
