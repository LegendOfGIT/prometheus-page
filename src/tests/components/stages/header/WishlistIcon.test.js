const React = require('react');
const renderer = require('react-test-renderer');

import WishlistIcon from '@/components/stages/header/WishlistIcon';

describe('WishlistIcon', () => {
    const getTestingProps = (props) => {
        return Object.assign(
            {},
            {
                setActiveModule: jest.fn()
            },
            props
        )
    };

    function assertSnapshot(props = getTestingProps()) {
        expect(renderer.create(<WishlistIcon {...props} />).toJSON()).toMatchSnapshot();
    }

    test('component renders without exploding', () => {
        assertSnapshot();
    });

    test('component renders active wishlist-icon if there is an item on the wishlist', () => {
        assertSnapshot(getTestingProps({
            wishlistHasItems: true
        }))
    });
});
