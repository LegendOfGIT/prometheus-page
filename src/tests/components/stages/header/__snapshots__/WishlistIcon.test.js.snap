// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WishlistIcon component renders active wishlist-icon if there is an item on the wishlist 1`] = `
<div
  className="wishlist"
>
  <img
    className="wishlist__icon"
    onClick={[Function]}
    src="/images/wishlist-selected.svg"
  />
  <h5
    className="wishlist__title"
  >
    Wishlist
  </h5>
</div>
`;

exports[`WishlistIcon component renders without exploding 1`] = `
<div
  className="wishlist"
>
  <img
    className="wishlist__icon"
    onClick={[Function]}
    src="/images/wishlist-neutral.svg"
  />
  <h5
    className="wishlist__title"
  >
    Wishlist
  </h5>
</div>
`;
